pkg/tests/ossm/ratelimit_test.go

```diff
@@ -33,82 +40,40 @@ var (
-	util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName)
-	util.KubeDeleteContents(meshNamespace, util.RunTemplate(rateLimitFilterYaml_template, Smcp))
-	time.Sleep(time.Second * 5)
-	util.KubeDeleteContents(meshNamespace, rateLimitSMCPPatch)
-	util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName)
-	redisDeploy.Uninstall()
-	bookinfoDeploy.Uninstall()
-}
-
 func TestRateLimiting(t *testing.T) {
-	test.NewTest(t).Id("T28").Groups(test.Full).NotRefactoredYet()
-
-	redisDeploy := examples.Redis{Namespace: "redis"}
-	bookinfo := examples.Bookinfo{Namespace: "bookinfo"}
-	bookinfo.Install(false)
-
-	defer cleanupRateLimiting(redisDeploy, bookinfo)
-
-	if err := redisDeploy.Install(); err != nil {
-		t.Fatal(err)
-	}
-	if _, err := util.Shell(`kubectl -n %s patch smcp/%s --type=merge --patch="%s"`, meshNamespace, smcpName, rateLimitSMCPPatch); err != nil {
-		t.Fatal(err)
-	}
-
-	if _, err := util.Shell(`oc -n %s wait --for condition=Ready smcp/%s --timeout 180s`, meshNamespace, smcpName); err != nil {
-		t.Fatal(err)
-	}
-
-	if err := util.CheckPodRunning(meshNamespace, "app=rls"); err != nil {
-		t.Fatalf("rls deployment not ready: %v", err)
-	}
-
-	if err := util.KubeApplyContents(meshNamespace, util.RunTemplate(rateLimitFilterYaml_template, Smcp)); err != nil {
-		t.Fatalf("error applying envoy filter: %v", err)
-	}
-	util.Shell(`kubectl -n %s get envoyfilter -o yaml > rrr.yaml`, meshNamespace)
-	// util.Log.Info(msg)
-
-	// Give some time to envoy filters apply
-	time.Sleep(time.Second * 5)
-
-	host, err := util.Shell("oc -n %s get route istio-ingressgateway -o jsonpath='{.spec.host}'", meshNamespace)
-	if err != nil {
-		t.Fatalf("error getting route hostname: %v", err)
-	}
-	host = strings.Trim(host, "'")
-
-	time.Sleep(time.Duration(20) * time.Second)
-
-	// Should work first time
-	checkProductPageResponseCode(t, host, "200")
-
-	// Should fail first time
-	checkProductPageResponseCode(t, host, "429")
-
-	// Should work again after 1 minute
-	time.Sleep(time.Second * 65)
-	checkProductPageResponseCode(t, host, "200")
-
-	// Should fail
-	// time.Sleep(time.Second * 20)
-	// checkProductPageResponseCode(t, host, "429")
-}
-
-func checkProductPageResponseCode(t *testing.T, host string, expectedCode string) {
-	t.Helper()
-
-	code, err := util.Shell("curl -s -o /dev/null -w '%%{http_code}' http://%s/productpage", host)
-	if err != nil {
-		t.Fatalf("error getting productpage: %v", err)
-	}
-	code = strings.Trim(code, "'")
-	if code != expectedCode {
-		t.Fatalf("expected status code %q got %q", expectedCode, code)
-	}
+	NewTest(t).Id("T28").Groups(Full).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		namespaces := []string{"bookinfo", "redis"}
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
```

Comment:
But for at least smcp we are always recreating the mesh namespace, no? I think we agree that always delete namespace to make sure that no configurations are left behind

Reply:
Then we need to change some tests because at least me always recreate meshNamespace too. Going to make the changes in another PR and submit them all in one, except this one